test/ability_flower_gift.c

```diff
@@ -0,0 +1,86 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim in harsh sunlight")
+{
+    GIVEN {
+        PLAYER(SPECIES_CHERRIM) { Ability(ABILITY_FLOWER_GIFT); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUNNY_DAY); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim back to normal when weather changes")
+{
+    GIVEN {
+        PLAYER(SPECIES_CHERRIM) { Ability(ABILITY_FLOWER_GIFT); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUNNY_DAY); }
+        TURN { MOVE(opponent, MOVE_RAIN_DANCE); }
+    } SCENE {
+        // transforms in sun
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+        // back to normal
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Flower Gift transforms Cherrim back to normal when its ability is suppressed")
+{
+    GIVEN {
+        PLAYER(SPECIES_CHERRIM) { Ability(ABILITY_FLOWER_GIFT); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_SUNNY_DAY); }
+        TURN { MOVE(opponent, MOVE_GASTRO_ACID); }
+    } SCENE {
+        // transforms in sun
+        ABILITY_POPUP(player, ABILITY_FLOWER_GIFT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+        // back to normal
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_FORM_CHANGE, player);
+        MESSAGE("Cherrim transformed!");
+    }
+}
+
+DOUBLE_BATTLE_TEST("Flower Gift increases the attack of Cherrim's ally by 1.5x", s16 damage)
```

Comment:
> During [harsh sunlight](https://bulbapedia.bulbagarden.net/wiki/Harsh_sunlight), Flower Gift increases the [Attack](https://bulbapedia.bulbagarden.net/wiki/Stat#Attack) and [Special Defense](https://bulbapedia.bulbagarden.net/wiki/Stat#Special_Defense) stats of the Pokémon with this Ability and its allies by 1.5×.

Bulbapedia's description imply the necessity of 3 more tests, one for its own Attack, its own Special Defense and the Ally's Special Defense.